File: Strings/pagina01/1222/1222.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){

    int total_palavras, max_linha, max_char;

    while (cin >> total_palavras >> max_linha >> max_char){

        vector<string> conto;
        for(int i = 0; i < total_palavras; i++){
            string palavra;
            cin >> palavra;
            conto.push_back(palavra);
        }
        int qnt_paginas = 0, ind = 0;

        while (ind < conto.size()) {
            int linhas = 0;
            while (linhas < max_linha && ind < conto.size()) {
                int linha_len = 0;
                while (ind < conto.size()) {
                    string palavra = conto[ind];
                    if (linha_len == 0) {
                        if (palavra.size() <= max_char) {
                            linha_len = palavra.size();
                            ind++;
                        } else {
                            break;
                        }
                    } else if (linha_len + 1 + palavra.size() <= max_char) {
                        linha_len += 1 + palavra.size();
                        ind++;
                    } else {
                        break;
                    }
                }
                linhas++;
            }
            qnt_paginas++;
        }

        cout << qnt_paginas << endl;
    }

    return 0;
}

File: Strings/pagina01/1222/1222.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        while (in.hasNextLine()) {
            String[] data = in.nextLine().split(" ");
            String[] conto = in.nextLine().split(" ");

            int max_linha = Integer.parseInt(data[1]);
            int max_char = Integer.parseInt(data[2]);
            int qnt_paginas = 0, ind = 0;

            while (ind < conto.length) {
                int linhas = 0;

                while (linhas < max_linha && ind < conto.length) {
                    int linha_len = 0;

                    while (ind < conto.length) {
                        String palavra = conto[ind];

                        if (linha_len == 0) {
                            if (palavra.length() <= max_char) {
                                linha_len = palavra.length();
                                ind++;
                            } else {
                                break;
                            }
                        } else if (linha_len + 1 + palavra.length() <= max_char) {
                            linha_len += 1 + palavra.length();
                            ind++;
                        } else {
                            break;
                        }
                    }

                    linhas++;
                }

                qnt_paginas++;
            }

            System.out.println(qnt_paginas);
        }

        in.close();
    }
}
